### PR TITLE
Backend auth

### PR DESCRIPTION
**Pull request by rgenge**
Authentication working( some parts are hard coded that will connect later on with user)
Added some variables in user

### CODE REVIEW

**Review by willianportilho — APPROVED** [review]

File: backend/.env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+NEST_API_CLIENT_ID="u-s4t2ud-941b73c259f6d553c86a6684f13a7532d9ee7991657547ad9e0e69ec1a37f16d"
```
Comment (koukunwlk) [inline backend/.env]: please, add the env file to the gitignore

File: backend/src/auth/42-auth/42.strategy.service.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  }
+  async validate(
+    accessToken: string,
+    refreshToken: string,
+    profile: any,
+    done: any,
+  ) {
+    const mfaSecret = await generateSecret(profile.id, 'Pong');
+    const secret = mfaSecret.secret;
+    console.log(secret)
```
Comment (koukunwlk) [inline backend/src/auth/42-auth/42.strategy.service.ts]: remove the log

File: backend/src/auth/42-auth/42.strategy.service.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    await this.userService.insertUser(
+      profile.username,
+      profile.token,
+      false,
+      profile.id,
+      profile.email,
+      profile.username,
+      secret
+    );
+    console.log(this.userService.getUser('acosta-a'))
```
Comment (koukunwlk) [inline backend/src/auth/42-auth/42.strategy.service.ts:44]: remove this too

File: backend/src/auth/auth.controller.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import { AuthService } from './auth.service';
+import { FortyTwoAuthGuard } from './42-auth/42.guards';
+import { JwtAuthGuard } from './jwt-auth/jwt-auth.guard';
+import { Response } from 'express';
+//import { UserService } from '../user/service/user.service'
+import { User } from '../user/domain/model/user/user';
+import { UserService } from '../user/service/user.service';
+//teste deletar
+
+const teste = new User({
```
Comment (koukunwlk) [inline backend/src/auth/auth.controller.ts]: After doing all your tests, you need to remove all unnecessary code to ensure the code quality

File: backend/src/auth/auth.controller.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  email: '',
+  username: '',
+  tfaSecret: '',
+});
+
+@Controller('auth')
+export class AuthController {
+  private savedTfaSecret: { secret: string }; // Variable to store TFA secret
+  constructor(
+    private authService: AuthService,
```
Comment (koukunwlk) [inline backend/src/auth/auth.controller.ts:22]: you must inject the services to use with nestjs feature

File: backend/src/auth/auth.controller.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  @Get('logout')
+  async logout(@Req() req: any, @Res() res: Response): Promise<void> {
+    res.clearCookie('token');
+    console.log('logout');
+    res.status(200).send({ message: 'Logout sucessful' });
+    //      return res.redirect(process.env.HOST_FRONT);
+  }
+  @Post('generate-2fa')
+  async generateTfaSecret(@Req() req) {
+    console.log('2fa Secret Generated');
```
Comment (koukunwlk) [inline backend/src/auth/auth.controller.ts]: remove

File: backend/src/auth/auth.controller.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    this.savedTfaSecret = { secret: '' };
+    const user = await this.userService.getUser('acosta-a'); //verificar com user
+    const tfa = await this.authService.generateTfaSecret(user.props.userId);
+    console.log(tfa.secret);
```
Comment (koukunwlk) [inline backend/src/auth/auth.controller.ts]: remove

File: backend/src/auth/auth.controller.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    console.log('logout');
+    res.status(200).send({ message: 'Logout sucessful' });
+    //      return res.redirect(process.env.HOST_FRONT);
+  }
+  @Post('generate-2fa')
+  async generateTfaSecret(@Req() req) {
+    console.log('2fa Secret Generated');
+    this.savedTfaSecret = { secret: '' };
+    const user = await this.userService.getUser('acosta-a'); //verificar com user
+    const tfa = await this.authService.generateTfaSecret(user.props.userId);
```
Comment (koukunwlk) [inline backend/src/auth/auth.controller.ts]: From hexagonal architecture, the controller cannot have our business logic, so we put the logic on a service and make all the necessary validations and return it for the controller

File: backend/src/auth/auth.controller.ts [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    console.log(tfa.secret);
+    this.savedTfaSecret.secret = tfa.secret;
+    return this.savedTfaSecret;
+  }
+
+  @Post('verify-2fa')
+  async verifyTfaSecret(@Req() req, @Body() body): Promise<void> {
+    console.log('testando verify:', req.user);
```
Comment (koukunwlk) [inline backend/src/auth/auth.controller.ts]: remove

File: backend/src/auth/auth.service.ts [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+import { User } from '../user/domain/model/user/user';
+import { generateSecret, verify } from '2fa-util';
+import { UsingJoinColumnIsNotAllowedError } from 'typeorm';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly userService: UserService,
+    private readonly jwtService: JwtService,
+    @Inject(USER_REPOSITORY_TOKEN)
```
Comment (koukunwlk) [inline backend/src/auth/auth.service.ts]: Why are you using a symbol as token?

File: backend/src/user/repository/user-typeorm.repository.ts [hunk 11]
```diff
@@ -27,6 +27,13 @@ export class UserTypeOrmRepository implements UserRepository {
   async insert(user: User): Promise<string> {
     const newUser = this.userRepository.create({
       nickname: user.getNickname(),
+      token: user.getToken(),
```
Comment (koukunwlk) [inline backend/src/user/repository/user-typeorm.repository.ts:30]: In the user model, create a function called toEntity that returns all fields to use like a object

like
```ts
class UserModel {
 private props: UserProps
 
 
 toEntity() {
  return this.props
 }
}

```

**Review by koukunwlk — CHANGES_REQUESTED** [review]

**Pull request merged**